src/test/java/org/jabref/logic/pdf/search/indexing/DocumentReaderTest.java

```diff
@@ -0,0 +1,102 @@
+package org.jabref.logic.pdf.search.indexing;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.pdf.search.SearchFieldConstants;
+
+import org.apache.lucene.document.Document;
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+public class DocumentReaderTest {
+
+    @Test(expected = IOException.class)
+    public void unknownFileTestShouldThrowIOException() throws IOException {
+        Path example = Paths.get("src/test/resources/pdfs/NOT_PRESENT.pdf");
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.of(example.toString()));
+
+        new DocumentReader(entry).readPdfContents();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void unknownFileTestShouldThrowIllegalArgumentException() throws IOException {
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.empty());
+
+        new DocumentReader(entry);
+    }
+
+    @Test
+    public void exampleTest() throws IOException {
+        Path example = Paths.get("src/test/resources/pdfs/example.pdf");
+
+        BibEntry entry = mock(BibEntry.class);
+        when(entry.getField(FieldName.FILE)).thenReturn(Optional.of(example.toString()));
+        when(entry.getCiteKeyOptional()).thenReturn(Optional.of("Example2017"));
+
+        Document doc = new DocumentReader(entry).readPdfContents();
+
+        assertEquals("Example2017", doc.get(SearchFieldConstants.KEY));
+        assertFalse(doc.get(SearchFieldConstants.CONTENT).isEmpty());
+        assertEquals("LaTeX with hyperref package", doc.get(SearchFieldConstants.CREATOR));
+    }
+
+    @Test
+    public void thesisExampleTest() throws IOException {
+        Path example = Paths.get("src/test/resources/pdfs/thesis-example.pdf");
+
+        BibEntry entry = mock(BibEntry.class);
```

Comment:
Well, the question is if this is supposed to be a unit or an integration test.
I have decided to write a unit test here to test the DocumentReader and ONLY the DocumentReader.

Of course in this test the difference to a small integration test is not much. Would propose to keep it like that and add further integration tests soon.